lib/options.ts

```diff
@@ -141,6 +158,9 @@ export type SliceResult =
 export interface HttpStack {
   createRequest(method: string, url: string): HttpRequest
   getName(): string
+  // Indicates whether this HTTP stack implementation supports progress events
+  // during upload. If false, stall detection will use overall transfer rate instead.
+  supportsProgressEvents?: () => boolean
```

Comment:
This function doesn't have to be optional as this PR would be released in a new major release, where we can make breaking changes.